public/app/features/profile/UserProfileEditPage.test.tsx

```diff
@@ -254,5 +325,99 @@ describe('UserProfileEditPage', () => {
+        await getTestContext();
+        const { extensionPointTabs } = getSelectors();
+        expect(extensionPointTabs()).not.toBeInTheDocument();
+      });
+
+      it('should group registered components into tabs', async () => {
+        await getTestContext({ extensions });
+        const { extensionPointTabs, extensionPointTab } = getSelectors();
+
+        const _assertTab = (tabName: string, isDefault = false) => {
+          const tab = extensionPointTab(tabName);
+          expect(tab).toBeInTheDocument();
+          expect(tab).toHaveAttribute('aria-selected', isDefault.toString());
+        };
+
+        expect(extensionPointTabs()).toBeInTheDocument();
+        _assertTab('Core', true);
+        _assertTab(tabOneName);
+        _assertTab(tabTwoName);
+      });
+
+      it('should change the active tab when a tab is clicked and update the "tab" query param', async () => {
+        const mockUpdateQueryParams = jest.fn();
+        mockUseQueryParams.useQueryParams = () => [{}, mockUpdateQueryParams];
+
+        await getTestContext({ extensions });
+        const { extensionPointTab } = getSelectors();
+
+        /**
+         * Tab one has two extension components registered against it, they'll both be registered in the same tab
+         * Tab two only has one extension component registered against it.
+         */
+        const tabOneContent1 = _createTabContent(ExtensionPointComponentId.One);
+        const tabOneContent2 = _createTabContent(ExtensionPointComponentId.Two);
+        const tabTwoContent = _createTabContent(ExtensionPointComponentId.Three);
+
+        // core should be the default content
+        expect(screen.queryByText(tabOneContent1)).toBeNull();
+        expect(screen.queryByText(tabOneContent2)).toBeNull();
+        expect(screen.queryByText(tabTwoContent)).toBeNull();
+
+        await userEvent.click(extensionPointTab(tabOneName));
+
+        expect(mockUpdateQueryParams).toHaveBeenCalledTimes(1);
+        expect(mockUpdateQueryParams).toHaveBeenCalledWith({ tab: tabOneName.toLowerCase() });
+        expect(screen.queryByText(tabOneContent1)).not.toBeNull();
+        expect(screen.queryByText(tabOneContent2)).not.toBeNull();
+        expect(screen.queryByText(tabTwoContent)).toBeNull();
+
+        mockUpdateQueryParams.mockClear();
+        await userEvent.click(extensionPointTab(tabTwoName));
+
+        expect(mockUpdateQueryParams).toHaveBeenCalledTimes(1);
+        expect(mockUpdateQueryParams).toHaveBeenCalledWith({ tab: tabTwoName.toLowerCase() });
+        expect(screen.queryByText(tabOneContent1)).toBeNull();
+        expect(screen.queryByText(tabOneContent2)).toBeNull();
+        expect(screen.queryByText(tabTwoContent)).not.toBeNull();
+      });
+
+      it.each([tabOneName, tabOneName.toUpperCase(), tabOneName.toLowerCase()])(
+        'should set the active tab based on the "tab" query param and be case-insensitive',
```

Comment:
why should it be case-insensitive? 🤔 afaik all our urls are supposed to be lower case

Reply:
I think I was overthinking edge-cases here. Your proposed approach of distinguishing between id (always lowercased) & title makes things simpler 👍